### ISSUE TITLE
Increase the default stack size for glibc-based applications

### ISSUE THREAD

**Issue by jxyang** [deislabs/mystikos#1110]
The test case is here: https://github.com/jxyang/mystikos/tree/websvc/solutions/websvc

To repro:
1) verify it runs fine without Mystikos:
```
docker build -t aaa .
docker run -p5000:5000 aaa & 
curl http://localhost:5000/weatherforecast
```
The client sends a http get request and should get back something like:

> [{"date":"2022-01-05T19:43:34.9427669+00:00","temperatureC":12,"temperatureF":53,"summary":"Sweltering"},{"date":"2022-01-06T19:43:34.9429125+00:00","temperatureC":7,"temperatureF":44,"summary":"Balmy"},{"date":"2022-01-07T19:43:34.942914+00:00","temperatureC":7,"temperatureF":44,"summary":"Bracing"},{"date":"2022-01-08T19:43:34.9429142+00:00","temperatureC":-11,"temperatureF":13,"summary":"Cool"},{"date":"2022-01-09T19:43:34.9429143+00:00","temperatureC":26,"temperatureF":78,"summary":"Balmy"}]

 2) Run it with Mystikos:
```
make && make run &
curl http://localhost:5000/weatherforecast
```
On server side, it acknowledges receiving the request with the below message, but just hangs without passing the control to the request handler.
```
Request starting HTTP/1.1 GET http://localhost:5000/weatherforecast
```

**Comment by jxyang** [comment]
@mikbras

**Comment by mikbras** [comment]
I see this line on my machine.

```
Microsoft.AspNetCore.Routing.EndpointMiddleware[0]
Executing endpoint 'websvc.Controllers.WeatherForecastController.Get (websvc)'
```

**Comment by anakrish** [comment]
Is the regression aspnet specific or is it some core issue with the sockets implementation?

**Comment by jxyang** [comment]
@anakrish I am inclined to aspnet specific as other network related tests passed in the pipelines.

**Comment by jxyang** [comment]
@mikbras and I discovered that:

1. Changing the app to be MUSL-based resolves the problem, and
2. Increasing the thread stack size to 1 mb resolves the problem.

This indicates there was a silent stack overflow, manifested as a hang, while executing with glibc, which typically demands bigger stack size.

Maybe it's time for us to revisit the default stack size issue given we are adopting glibc as the default CRT and moving to Icelake when has more generous EPC memory.

**Comment by paulcallen** [comment]
during packaging we can warn if stack size is too small in config:
- glibc is used
- .net is used
- python is used

At run-time when  config is used. Warn during start-up that stack size may not be big enough.

At run-time when no config is used, still warn about stack size on startup if not big enough.

**Comment by mikbras** [comment]
Fixed by PR #1132.
